File: Daily Challanges/2026-05-May/2026-05-29.py
from math import gcd


def get_wider_aspect_ratio(dim1, dim2):
    w1, h1 = map(int, dim1.split("x"))
    w2, h2 = map(int, dim2.split("x"))

    if w1 * h2 >= w2 * h1:
        w, h = w1, h1
    else:
        w, h = w2, h2

    g = gcd(w, h)
    return f"{w // g}:{h // g}"


print(get_wider_aspect_ratio("1920x1080", "800x600"))
print(get_wider_aspect_ratio("1080x1350", "2048x1536"))
print(get_wider_aspect_ratio("640x480", "2440x1220"))

File: Daily Challanges/2026-05-May/2026-05-29.js
function getWiderAspectRatio(dim1, dim2) {
  const [w1, h1] = dim1.split("x").map(Number);
  const [w2, h2] = dim2.split("x").map(Number);

  let w, h;

  if (w1 * h2 >= w2 * h1) {
    w = w1;
    h = h1;
  } else {
    w = w2;
    h = h2;
  }

  const gcd = (a, b) => {
    while (b !== 0) {
      [a, b] = [b, a % b];
    }
    return a;
  };

  const g = gcd(w, h);
  return `${w / g}:${h / g}`;
}

console.log(getWiderAspectRatio("1920x1080", "800x600"));
console.log(getWiderAspectRatio("1080x1350", "2048x1536"));
console.log(getWiderAspectRatio("640x480", "2440x1220"));
